pyspecProcScripts/correlation_alignment.py

```diff
@@ -20,117 +22,141 @@ def correl_align(s, align_phases=False,tol=1e-4,indirect_dim='indirect',fig_titl
                 maximize their correlation amongst each other, thereby aligning
-                them
+                them.
     sigma:      float
-                the width of the Gaussian function used to frequency filter
+                The width of the Gaussian function used to frequency filter
                 the data in the calculation of the correlation function.
     """
-    ph1_len = len(s.getaxis('ph1'))
-    ph2_len = len(s.getaxis('ph2'))
-    N = ndshape(s)[indirect_dim]
-    sig_energy = (abs(s)**2).data.sum().item() / N
-    if fl is not None:
-        fl.next('before correlation\nsig. energy=%g'%sig_energy + fig_title)
-        fl.image(s.C.setaxis('vd','#').set_units('vd','scan #'),human_units=False)
+
+    logging.info(strm("Applying the correlation routine"))
+    if avg_dim:
+        phcycdims = [j for j in s_orig.dimlabels if j.startswith('ph')]
+        indirect = set(s_orig.dimlabels)-set(phcycdims)-set([direct])
+        indirect = [j for j in s_orig.dimlabels if j in indirect]
+        avg_dim_len = len(s_orig.getaxis(avg_dim))
+        s_orig.smoosh(indirect)
+    for j in signal_pathway.keys():
+        assert not s_orig.get_ft_prop(j), str(j)+" must not be in the coherence domain"
+    signal_keys = list(signal_pathway)
+    signal_values = list(signal_pathway.values())
+    ph_len = {j:ndshape(s_orig)[j] for j in signal_pathway.keys()}
+    N = ndshape(s_orig)[indirect_dim]
+    sig_energy = (abs(s_orig)**2).data.sum().item() / N
+    if fl:
+        fl.push_marker()
+        fig_forlist, ax_list = plt.subplots(1, 5, figsize=(7,7))
+        fl.next("Correlation Diagnostics")
+        fig_forlist.suptitle(" ".join(["Correlation Diagnostic"] + [j for j in [fl.basename] if j is not None]))
+        fl.image(s_orig.C.setaxis(indirect_dim,'#').set_units(indirect_dim,'scan #'),ax=ax_list[0],human_units=False)
+        ax_list[0].set_title('before correlation\nsig. energy=%g'%sig_energy)
     energy_diff = 1.
     i = 0
     energy_vals = []
-    this_E = (abs(s.C.sum(indirect_dim))**2).data.sum().item() / N**2
+    this_E = (abs(s_orig.C.sum(indirect_dim))**2).data.sum().item() / N**2
     energy_vals.append(this_E / sig_energy)
     last_E = None
-    for_nu_center =s.C
-    for_nu_center.ft(['ph1','ph2'])
-    for_nu_center = for_nu_center['ph1',ph1_selection]['ph2',ph2_selection]
-    nu_center = for_nu_center.mean(indirect_dim).C.argmax('t2')
+    for_nu_center =s_orig.C
+    for_nu_center.ft(list(signal_pathway))
+    for x in range(len(signal_keys)):
+        for_nu_center = for_nu_center[signal_keys[x],signal_values[x]]
+    nu_center = for_nu_center.mean(indirect_dim).C.argmax(direct)
     logging.info(strm("Center frequency", nu_center))
     for my_iter in range(100):
         i += 1
         logging.info(strm("*** *** ***"))
         logging.info(strm("CORRELATION ALIGNMENT ITERATION NO. ",i))
         logging.info(strm("*** *** ***"))
+        s_orig.ift(direct)
```

Comment:
why?/what is this for?

Reply:
you had added the ift --> I just changed the name from `s` to `s_orig` to clarify a bit more.

Reply:
yes, this is just line 82 below in read (from b74fba) --> in addition to changing the name, I also moved it up and changed the fact that you were applying the ph0 to the original data instead of the copy

Reply:
if you're **rather** saying it would be smarter to just demand signal in the time domain, so that you don't ift the `s_orig` at the beginning and ft it at the end, then by all means, please do that!  You would, though, want to make sure that you pass an assert statement at the beginning, and that the signal energy calculation looks reasonable.

Reply:
okay no that makes sense! I was just wondering what it was doing is all but it makes sense now!